addons/l10n_ch/models/account_invoice.py

```diff
@@ -345,3 +347,38 @@ def space_qrr_reference(self, qrr_ref):
             i -= 5
 
         return spaced_qrr_ref
+
+    def action_create_batch_qr(self):
+        return self.env.ref('l10n_ch.l10n_ch_qr_report').report_action(self)
+
+    def qr_batch_action(self):
+        for rec in self:
+            if not rec.move_type == 'out_invoice':
+                raise UserError(_("Only invoices can be QR-printed."))
```

Comment:
```py
if any(x.move_type != 'out_invoice' for x in self):
    raise UserError(_("Only invoices can be QR-printed."))
```
What about `out_refund` / `out_receipt` ?
Shouldn't be `Only customer invoices` then?